Libs/DICOM/Widgets/ctkDICOMBrowser.cpp

```diff
@@ -657,3 +667,177 @@ void ctkDICOMBrowser::onModelSelected(const QItemSelection &item1, const QItemSe
+    + QString::number(numPatients)
+    + QString(" selected patients");
+  QAction *deleteAction = new QAction(deleteString, patientsMenu);
+
+  patientsMenu->addAction(deleteAction);
+
+  // the table took care of mapping it to a global position so that the
+  // menu will pop up at the correct place over this table.
+  QAction *selectedAction = patientsMenu->exec(point);
+
+  if (selectedAction == deleteAction
+      && this->confirmDeleteSelectedIUDs(selectedPatientsUIDs))
+    {
+    qDebug() << "Deleting " << numPatients << " patients";
+    foreach (const QString& uid, selectedPatientsUIDs)
+      {
+      d->DICOMDatabase->removePatient(uid);
+      }
+    }
+}
+
+//----------------------------------------------------------------------------
+void ctkDICOMBrowser::onStudiesRightClicked(const QPoint &point)
+{
+  Q_D(ctkDICOMBrowser);
+
+  // get the list of studies that are selected
+  QStringList selectedStudiesUIDs = d->dicomTableManager->currentStudiesSelection();
+  int numStudies = selectedStudiesUIDs.size();
+  if (numStudies == 0)
+    {
+    qDebug() << "No studies selected!";
+    return;
+    }
+
+  QMenu *studiesMenu = new QMenu(d->dicomTableManager);
+
+  QString deleteString = QString("Delete ")
+    + QString::number(numStudies)
+    + QString(" selected studies");
+  QAction *deleteAction = new QAction(deleteString, studiesMenu);
+
+  studiesMenu->addAction(deleteAction);
+
+  // the table took care of mapping it to a global position so that the
+  // menu will pop up at the correct place over this table.
+  QAction *selectedAction = studiesMenu->exec(point);
+
+  if (selectedAction == deleteAction
+      && this->confirmDeleteSelectedIUDs(selectedStudiesUIDs))
+    {
+    foreach (const QString& uid, selectedStudiesUIDs)
+      {
+      d->DICOMDatabase->removeStudy(uid);
+      }
+    }
+}
+
+//----------------------------------------------------------------------------
+void ctkDICOMBrowser::onSeriesRightClicked(const QPoint &point)
+{
+  Q_D(ctkDICOMBrowser);
+
+  // get the list of series that are selected
+  QStringList selectedSeriesUIDs = d->dicomTableManager->currentSeriesSelection();
+  int numSeries = selectedSeriesUIDs.size();
+  if (numSeries == 0)
+    {
+    qDebug() << "No series selected!";
+    return;
+    }
+
+  QMenu *seriesMenu = new QMenu(d->dicomTableManager);
+
+  QString deleteString = QString("Delete ")
+    + QString::number(numSeries)
+    + QString(" selected series");
+  QAction *deleteAction = new QAction(deleteString, seriesMenu);
+
+  seriesMenu->addAction(deleteAction);
+
+  // the table took care of mapping it to a global position so that the
+  // menu will pop up at the correct place over this table.
+  QAction *selectedAction = seriesMenu->exec(point);
+
+  if (selectedAction == deleteAction
+      && this->confirmDeleteSelectedIUDs(selectedSeriesUIDs))
+    {
+    foreach (const QString& uid, selectedSeriesUIDs)
+      {
+      d->DICOMDatabase->removeSeries(uid);
+      }
+    }
+}
```

Comment:
would it make sense to refactor `onPatientsRightClicked`, `onStudiesRightClicked` and `onSeriesRightClicked` so that they call the same function ? 

To do so, you may have to add a function `currentInstancesSelection(int instanceType)` to the table manager ...